js/module.js

```diff
@@ -378,23 +378,24 @@ const Module = Class.extend({
 	 * @param {Function} callback Called when the animation is done.
 	 * @param {object} [options] Optional settings for the hide method.
 	 */
-	hide: function (speed, callback, options) {
+	hide: function (speed, callback, options = {}) {
+		let usedCallback = callback || function () {};
+		let usedOptions = options;
+
 		if (typeof callback === "object") {
-			options = callback;
-			callback = function () {};
+			Log.error("Parameter mismatch in module.hide: callback is not an optional parameter!");
+			usedOptions = callback;
+			usedCallback = function () {};
 		}
```

Comment:
no idea why this was handled this way from the beginning by @MichMich 

callback was never optional, but now some modules might be relying on this function call the way it is. Thats why I added an error warning here so that it might help people write better modules ...